src/harness/dynamicUnittestsImpl.ts

```diff
@@ -0,0 +1,73 @@
+
+namespace DynamicUnittests {
+    export const enum TestType {
+        Tsbuild = "tsbuild",
+        TsbuildWatch = "tsbuildWatch",
+        Tsc = "tsc",
+        TscWatch = "tscWatch",
+        Tsserver = "tsserver",
+    }
+    export function runTests(basePath: string, fileName: string, dynamicTestName: string) {
+        const content = Harness.IO.readFile(fileName)!;
+
+        // Give file paths an absolute path for the virtual file system
+        const absoluteBasePath = ts.combinePaths(Harness.virtualFileSystemRoot, basePath);
+        const absoluteFileName = ts.combinePaths(Harness.virtualFileSystemRoot, fileName);
+
+        // Parse out the files and their metadata
+        // const testData = parseTestData(absoluteBasePath, content, absoluteFileName);
+        // const state = new TestState(absoluteFileName, absoluteBasePath, testType, testData);
+        const actualFileName = Harness.IO.resolvePath(fileName) || absoluteFileName;
+        const output = ts.transpileModule(content, { reportDiagnostics: true, fileName: actualFileName, compilerOptions: { target: ts.ScriptTarget.ES2015, inlineSourceMap: true, inlineSources: true } });
+        if (output.diagnostics?.length) {
+            throw new Error(`Syntax error in ${absoluteBasePath}: ${ts.formatDiagnostics(output.diagnostics, {
+                getNewLine: () => "\n",
+                getCanonicalFileName: ts.createGetCanonicalFileName(Harness.IO.useCaseSensitiveFileNames()),
+                getCurrentDirectory: () => ts.getDirectoryPath(absoluteBasePath)
+            })}`);
+        }
+        runCode(output.outputText, actualFileName, dynamicTestName);
+    }
+
+    function runCode(code: string, fileName: string, dynamicTestName: string): void {

```

Comment:
Most of this is just like fourslash except this builds run.d.ts and these tests reference them so they can do what unittests do.